### ISSUE TITLE
Rule proposal: max-statements

### ISSUE THREAD

**Issue by Alanrah** [typescript-eslint/typescript-eslint#4585]
### Description

The rule enforces a maximum number of statements allowed in function blocks. like https://eslint.org/docs/rules/max-statements

### Fail

```ts
// when max-statements is ["error", 10], the code below should be error.
function test() {
  const foo1 = 1;
  const foo2 = 2;
  const foo3 = 3;
  const foo4 = 4;
  const foo5 = 5;
  const foo6 = 6;
  const foo7 = 7;
  const foo8 = 8;
  const foo9 = 9;
  const foo10 = 10;

  const foo11 = 11;
}
```

### Pass

```ts
// when max-statements is ["error", 10], the code below should be ok.
function test() {
  const foo1 = 1;
  const foo2 = 2;
  const foo3 = 3;
  const foo4 = 4;
  const foo5 = 5;
  const foo6 = 6;
  const foo7 = 7;
  const foo8 = 8;
  const foo9 = 9;
  const foo10 = 10;
}
```

**Comment by bradzacher** [comment]
I'm not sure I understand... What is the point of an extension rule here? What would we be adding on? Could you please elaborate?

**Comment by Alanrah** [comment]
Sorry. More clearly, I need a rule to check number of statements in a function. When the number of a function statements exceeds maximum, an error can be reported.

**Comment by bradzacher** [comment]
Just use eslint's `max-statements` rule then? Again I'm not sure what the ask is here.

**Comment by Alanrah** [comment]
I tried it, it doesn't seem to work, I will try again, thanks for the reply
